charts/presto/templates/configmap-coordinator.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-coordinator
+  labels:
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Values.image.tag | default .Chart.AppVersion | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+data:
+  node.properties: |-
+    node.environment={{ .Values.node.environment }}
+    node.data-dir={{ .Values.node.dataDir }}
```

Comment:
The `node.location=` is missing here same goes for worker. I assume that `node.id` is generated on the run?

Reply:
Cannot find `node.location` in the docs nor source code, except [here](https://github.com/prestodb/presto/blob/fb0c4dc0b4c7b1b570501ad6054113b5d538f79e/presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/NativeExecutionNodeConfig.java#L28). Is it common Presto config or is used by native executor only?

Yes, `node.id` is generated during startup, so
```
SELECT * FROM system.runtime.nodes;
+------------------------------------+----------------------+-------------+-----------+------+
|node_id                             |http_uri              |node_version |coordinator|state |
+------------------------------------+----------------------+-------------+-----------+------+
|82c21439-ecd7-46c2-9a1d-3714f44abe5a|http://172.17.0.6:8080|0.278-6156edd|false      |active|
|d0a8906c-1665-483f-92b3-f6ab0c128028|http://172.17.0.7:8080|0.278-6156edd|false      |active|
|f0bb7f12-9aec-4ed6-82bd-7071b896e07e|http://172.17.0.8:8080|0.278-6156edd|true       |active|
+------------------------------------+----------------------+-------------+-----------+------+
```

Reply:
The main part is here:
Add release notes for 0.251 #15923: https://github.com/prestodb/presto/pull/15923
Use node location instead of nodeId to validate nodes #15875: https://github.com/prestodb/presto/pull/15875

Reply:
No having this setting overwritten can result in hard to debug behavior. Same goes with `node.version`, if there is a miss match the node won't be able to connect (we had this issue when connecting earlier versions of prestissimo).

Reply:
I didn't face the issues you mentioned and not sure I fully understand `node.location` behavior. So I would keep it as is for now and wait for other comments.

Reply:
So let me explain this and cite main goal of `node.location` existance:

>#15875: This PR has code changes to use node.location instead of node.id for worker node validation. Until now we are sending nodeId to NodeStatusService.isAllowed() method to figure out if a node is allowed to register with DiscoveryNodeManager. But with this PR, we will be sending nodeLocation to NodeStatusService to figure out if this node is allowed to register or not.

Avoiding going in to the rabbit hole - the `node.location` should be used for allowed/disallowed nodes distinguishing as `node.id` have a tendency to change like in case of `pod` being reloaded inside `Kubernetes`. If this is not provided for default the `/node.id` is used but this case is discouraged.

Reply:
Hm... This probably should be documented, for example, [here](https://prestodb.io/docs/current/installation/deployment.html#node-properties).
Do you have any suggestions for default `node.location` value in the helm chart?

Reply:
It is not documented at all and should be as can cause issues. I would set it to something like `test-location` or `presto.local` (analogy to k8s `cluster.local`).

Reply:
https://github.com/prestodb/presto/pull/18700/commits/dcd27534a87d8b8a0ad463d728937281a7318e75 - I've added `node.location` property with default value `presto.local`  which also matches default Ingress host.
Would you mind to create an issue or maybe PR to document `node.location` property as you know more context about it?